packages/base/src/mainview/TemporalSlider.tsx

```diff
@@ -0,0 +1,430 @@
+    model.sharedLayersChanged.connect(handleLayerChange);
+
+    return () => {
+      model.clientStateChanged.disconnect(handleClientStateChanged);
+      model.sharedLayersChanged.disconnect(handleLayerChange);
+      removeFilter();
+      if (intervalRef.current) {
+        clearInterval(intervalRef.current);
+      }
+    };
+  }, []);
+
+  useEffect(() => {
+    layerIdRef.current = layerId;
+  }, [layerId]);
+
+  useEffect(() => {
+    const featuresForSelect = [];
+
+    // We only want to show features that could be time values
+    for (const [key, set] of Object.entries(featureProperties)) {
+      let checkIfDateIsValid = false;
+      const checkValue = set.values().next().value;
+
+      // We only want to look at strings and whole numbers
+      // ? Is there a better way to check if number values are valid timestamps?
+      // ! QGIS doesn't actually support number values for their time thing
+      const isString = typeof checkValue === 'string';
+      const isNumber =
+        typeof checkValue === 'number' && Number.isInteger(checkValue);
+      if (!isString && !isNumber) {
+        continue;
+      }
+
+      checkIfDateIsValid = isValid(toDate(checkValue));
+      if (!checkIfDateIsValid) {
+        continue;
+      }
+
+      if (checkValue && checkIfDateIsValid) {
+        featuresForSelect.push(key);
+      }
+    }
+
+    setValidFeatures(featuresForSelect);
+    setSelectedFeature(featuresForSelect[0]);
+  }, [featureProperties]);
+
+  useEffect(() => {
+    if (!selectedFeature) {
+      return;
+    }
+
+    // Get the values from the selected feature
+    const values: any[] = Array.from(featureProperties[selectedFeature]);
+
+    // Check the type of the first element
+    const isString = typeof values[0] === 'string';
+    let convertedValues;
+
+    if (isString) {
+      const inferredFormat = inferDateFormat(values[0]);
+      if (!inferredFormat) {
+        console.log('Date string has an unsupported format');
+        return;
+      }
+
+      setInferredDateFormat(inferredFormat);
+
+      convertedValues = values.map(value => Date.parse(value)); // Convert all strings to milliseconds
+      setInferredDateFormat(inferredFormat);
+    } else {
+      convertedValues = values; // Keep numbers as they are
+    }
+
+    // Calculate min and max
+    const min = Math.min(...convertedValues);
+    const max = Math.max(...convertedValues);
+
+    // Get valid step options
+    const filteredSteps = Object.fromEntries(
+      Object.entries(stepMap).filter(([_, val]) => val < max - min)
+    );
+
+    setCurrentValue(min);
+    setMinMax({ min, max });
+    setRange({ start: min, end: min + step });
+    setValidSteps(filteredSteps);
+    setStep(Object.values(filteredSteps)[0]);
+
+    model.addFeatureAsMs(layerId, selectedFeature);
+  }, [selectedFeature]);
+
+  // Infer the date format from a date string
+  const inferDateFormat = (dateString: string): string | null => {
```

Comment:
Nit: Infer would indicate we're looking at `dateString` and using its content to determine the date format. Instead, we're finding it by trial-and-error; the first format that parses is chosen. To me, "infer" is an implementation detail that is better omitted from the function name. How about `determineDateFormat` or `dateFormatFromDateString`?